### PR TITLE
Fix popups appearing at 0,0

### LINKED ISSUE #7749: Messages are appearing at 0, 0 again.
## Description
image

Only caught because we disabled random station offset. Remind me to make that into a coinflip so sometimes the station is at 0, 0.

### PR DESCRIPTION
**Pull request by ElectroJr**
Problem was probably just because of entities leaving PVS and being sent to 0,0 in null-space. Should fix #7749 but I haven't really tested thoroughly.

### CODE REVIEW

File: Content.Client/Popups/PopupSystem.cs [hunk 1]
```diff
@@ -43,7 +42,10 @@ public void PopupCursor(string message)
 
         public void PopupCoordinates(string message, EntityCoordinates coordinates)
         {
-            PopupMessage(message, _eyeManager.CoordinatesToScreen(coordinates));
+            var mapCoords = coordinates.ToMap(EntityManager);
+            if (_eyeManager.CurrentMap != mapCoords.MapId)
```
Comment (PaulRitter) [inline Content.Client/Popups/PopupSystem.cs:46]: the client can already see multiple maps at once using viewsubscriptions. i dont have time to figure out how you can check which maps are seen atm but i know this is wrong

File: Content.Client/Popups/PopupSystem.cs [hunk 2]
```diff
@@ -52,6 +54,9 @@ public void PopupEntity(string message, EntityUid uid)
                 return;
 
             var transform = EntityManager.GetComponent<TransformComponent>(uid);
+            if (_eyeManager.CurrentMap != transform.MapID)
```
Comment (PaulRitter) [inline Content.Client/Popups/PopupSystem.cs:57]: see above

File: Content.Client/Popups/PopupSystem.cs [hunk 3]
```diff
@@ -205,15 +210,21 @@ protected override void FrameUpdate(FrameEventArgs eventArgs)
                 TotalTime += eventArgs.DeltaSeconds;
 
                 Vector2 position;
-                if (Entity == null)
+                if (Entity == null && InitialPos.MapId == _eyeManager.CurrentMap)
```
Comment (PaulRitter) [inline Content.Client/Popups/PopupSystem.cs:213]: see above

File: Content.Client/Popups/PopupSystem.cs [hunk 4]
```diff
@@ -205,15 +210,21 @@ protected override void FrameUpdate(FrameEventArgs eventArgs)
...
+                {
+                    Visible = true;
                     position = _eyeManager.WorldToScreen(InitialPos.Position) / UIScale - DesiredSize / 2;
-                else if (_entityManager.TryGetComponent(Entity.Value, out TransformComponent xform))
+                }
+                else if (_entityManager.TryGetComponent(Entity, out TransformComponent xform) && xform.MapID == _eyeManager.CurrentMap)
```
Comment (PaulRitter) [inline Content.Client/Popups/PopupSystem.cs:218]: see above

**Review by PaulRitter — CHANGES_REQUESTED** [review]

**Comment by metalgearsloth** [comment]
@PaulRitter A whole bunch of stuff is broken for cameras / multi-viewports which needs fixing anyway, see https://github.com/space-wizards/space-station-14/issues/8154

**Comment by PaulRitter** [comment]
> @PaulRitter A whole bunch of stuff is broken for cameras / multi-viewports which needs fixing anyway, see #8154

image

**Comment by metalgearsloth** [comment]
Merging coz I wanna stop getting pinged on it and I'm too lazy to revert the log.

**Pull request merged**